src/smt/theory_str_regex.cpp

```diff
@@ -902,7 +902,11 @@ namespace smt {
             return true;
         } else if (u.re.is_complement(re)) {
             // TODO can we do better?
-                    return false;
+            return false;
+        } else if (u.re.is_intersection(re)) {
+            return false;
+        } else if (u.re.is_complement(re)) {
```

Comment:
I think this one is redundant given the one right above isn't it?

Reply:
You're right, but eventually, these will have two different heuristics for length, and I wanted to capture that by separating them out.